firestore.rules

```diff
@@ -20,20 +20,76 @@ service cloud.firestore {
       return collection == "user";
     }
 
-    match /users/{userId} {
+    match /users/{id} {
       // Allow a request if the user ID in the path is 12345
       // e.g. /projects/projectId/databases/(default)/documents/users/12345
       allow read, update: if request.path[6] == request.auth.uid
     }
 
-    match /communityGroups/{communityGroupId} {
+    match /summerMissions/{id} {
+      // Data validation for required fields
+      allow write: if request.resource.data.keys().hasAll(['name', 'location', 'startDate', 'endDate'])
+                   && request.resource.data.name is string
+                   && request.resource.data.location is string
+                   && request.resource.data.startDate is timestamp
+                   && request.resource.data.endDate is timestamp
+                   && isAdmin();
+    }
+
+    match /campuses/{id} {
+      // Data validation for required fields
+      allow write: if request.resource.data.keys().hasAll(['name', 'location'])
+                   && request.resource.data.name is string
+                   && request.resource.data.location is string
+                   && isAdmin();
+    }
+
+    match /communityGroups/{id} {
       allow update: if isLeader();
```

Comment:
You are going to need explicit rules here also. the update rules should be mostly the same as the write rules with the condition that the leader info can't be changed.